compiler/rustc_errors/src/lib.rs

```diff
@@ -7,6 +7,7 @@
 #![feature(backtrace)]
 #![feature(if_let_guard)]
 #![feature(let_else)]
+#![feature(never_type)]
```

Comment:
I don't know if it's worthwhile introducing more cases of `never_type` since T-lang has discussed cutting back some of the specialness of the type.